source-mongodb/pull.go

```diff
@@ -320,7 +341,7 @@ func (c *capture) BackfillCollection(ctx context.Context, client *mongo.Client,
 		state.BackfillStartedAt = time.Now()
 	}
 
-	var opts = options.Find().SetSort(bson.D{{idProperty, SortAscending}})
+	var opts = options.Find().SetSort(bson.D{{idProperty, SortAscending}}).SetBatchSize(BACKFILL_BATCH_SIZE)
```

Comment:
We'll still return the results for the entire collection here right and not just `BACKFILL_BATCH_SIZE` docs? From a skim of the MongoDB docs I'm ~95% sure if that's how it works but wanted to ask.

More importantly: Do you have a good idea of what this does to connector memory? I'm wondering if it gets all `BACKFILL_BATCH_SIZE` docs at once and has to store them all in memory, or if the client is smarter than that. It could be a problem if the documents are very large to get that many at once.

Reply:
> We'll still return the results for the entire collection here right and not just BACKFILL_BATCH_SIZE docs? From a skim of the MongoDB docs I'm ~95% sure if that's how it works but wanted to ask.

Yes I've verified that we do still capture all of the documents, the batch size is explained [here](https://www.mongodb.com/docs/manual/reference/command/find/) and [here](https://www.mongodb.com/docs/manual/reference/method/cursor.batchSize/). The first one explains what happens with `find` command itself, and the second one explains what happens when you set that option on the `cursor` that you get from the command. The driver [does both](https://github.com/mongodb/mongo-go-driver/blob/v1.11.0/mongo/collection.go#L1235-L1237).

The behaviour you mentioned happens if you set `singleBatch` to true, which we don't, and by default that is `false`.